collect_app/src/main/java/org/odk/collect/android/formmanagement/ServerFormDownloader.java

```diff
@@ -301,41 +311,25 @@ private Uri saveNewForm(Map<String, String> formInfo, File formFile, String medi
          * object representing the downloaded file.
          */
         FileResult downloadXform(String formName, String url, FormDownloaderListener stateListener, File tempDir, String formsDirPath) throws FormSourceException, IOException, InterruptedException {
-            // clean up friendly form name...
-            String rootName = FormNameUtils.formatFilenameFromFormName(formName);
-
-            // proposed name of xml file...
-            String path = formsDirPath + File.separator + rootName + ".xml";
-            int i = 2;
-            File tempFormFile = new File(path);
-            while (tempFormFile.exists()) {
-                path = formsDirPath + File.separator + rootName + "_" + i + ".xml";
-                tempFormFile = new File(path);
-                i++;
-            }
-
-            InputStream file = formSource.fetchForm(url);
-            writeFile(tempFormFile, stateListener, file, tempDir);
```

Comment:
This was the key problem: `tempFormFile` is defined using the `formsDirPath` instead of `tempDir`.